ldap_check.sh

```diff
@@ -0,0 +1,300 @@
+echo "	/support.jumpcloud.com/knowledgebase/articles/413926-jumpcloud-s-"
+echo "	hosted-ldap-service						"
+echo ""
+echo "Step 2. Pick a port using option 2 or 3				"
+echo ""
+echo "Step 3. Option 4 will allow you to connect to LDAP and search  	"
+echo "using the defined account variables				"
+echo ""
+echo "Option 5 will list common strings used by the applications	"
+echo "that will athenticate to JumpCloud LDAP.				"
+echo ""
+echo "Option 9 displays this help screen				"
+echo ""
+echo "Option 0 will exit the utility					"
+}
+
+
+#
+# ldapsearch
+#
+
+# Check for ldapsearch command
+
+ldapsearch=`which ldapsearch`
+
+which ldapsearch &>/dev/null;
+
+if [ $? != 0 ]
+	then
+		echo "ldapsearch is not in your path or not installed.  Please correct and rerun this script";
+		exit 1;
+fi
+
+
+# Check account vars have been entered
+
+check_ldap_config() {
+
+if [ -z "$port" ]
+        then
+                echo "Port Undefined, using default 636";
+                port=636;
+                read -rsp $'press key to continue \n' -n1 key
+fi
+
+if [ "$port" = "389 -ZZ" ]
+        then
+                uri=ldap
+        else
+                uri=ldaps
+fi
+
+if [ -z "$user" ]
+        then
+                echo -e "\nAccount variables undefined, enter information and reselect\n";
+                read_account;
+fi
+}
+
+# Define ldapsearch 
+
+ldsearch() {
+
+check_ldap_config
+
+$ldapsearch -H ${uri}://ldap.jumpcloud.com:${port} -x -b "ou=Users,o=${oid},dc=jumpcloud,dc=com" -D "uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com" -w "${pass}" "(objectClass=${search_param})" | less
+
+}
+
+#
+# Display common configuration strings
+#
+
+echo_ldap_config() {
+
+echo "###################################################################"
+echo "#                Common LDAP Configuration Settings               #"
+echo "###################################################################"
+echo ""
+echo "### URI/LDAP Server ###"
+echo ""
+echo "ldaps://ldap.jumpcloud.com:636"
+echo "ldap://ldap.jumpcloud.com:389"
+echo ""
+echo "### BIND DN ###"
+echo ""
+echo "uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com"
+echo ""
+echo "### Search DN/Base Search DN ###"
+echo ""
+echo "ou=Users,o=${oid},dc=jumpcloud,dc=com"
+echo ""
+echo "### LDAP Search Example ###"
+echo ""
+echo "$ldapsearch -H ${uri}://ldap.jumpcloud.com:${port} -x -b \"ou=Users,o=${oid},dc=jumpcloud,dc=com\" -D \"uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com\" -w \"${pass}\" \"(objectClass=inetOrgPerson)\""
+echo ""
+echo "**If the above ldapsearch command results in invalid credentials, and the password contains special characters (!,@,#,etc...), replace the double quotes (\") around the password with single quotes (') and retry"
+echo ""
+}
+
+# Display the settings
+
+ldap_config() {
+
+check_ldap_config
+echo_ldap_config | less
+
+}
+
+#
+# read in oid, ldap service account and password
+#
+
+# read username, check for null input
+
+read_user() {
```

Comment:
This and the read_pass function could be turned into a single function that simply takes a string like "LDAP Binding User account name" and returns the result.